### ISSUE TITLE
benchmarks are meaningless

### ISSUE THREAD

**Issue by japajoe** [boomzero/http_server#2]
First of all this is not really an issue and I think you have an interesting project. Your benchmarks however are a bit misleading in my opinion. For example the hardware you use to run your server does matter a lot. Another factor is how thorough you parse an http request, and how much work is involved in preparing the response. I've been working on my own little project and benchmark results vary greatly depending on the hardware I use. On one machine my library was able to handle over 5000 (https) requests per second, but on another it's only about 1200. The numbers also seem to go down quite a bit if you use https instead of http. Just wanted to say that you shouldn't rely too much on the benchmarks without taking different things into consideration. The reason I write this is because I was curious about the performance of my own library, but I realized this is a bit difficult to measure. Don't let me discourage you though, that is not my intention. Happy programming :)

**Comment by boomzero** [comment]
First of all, thank you for your interest in this project.
I actually used a Raspberry Pi for my benchmarks. I thought that since this is probably the worst possible hardware anyone could use, it would be the least amount of performance one could expect from `http-server` (You are definitely right, hardware does matter a lot, especially when you are dealing with raw sockets, like I am). As for how I parse through my request,  I believe my benchmarks reflect what `http-server` does for most requests, after all, I am just pulling up a file using `ifstream`, and sending it through. (Of course, I may be wrong. I often am!) 
I also had lightpd running on this raspberrypi, so I ran `ab` on it to, and got around `2900` requests per second(http).
This is why I state lightpd is slower.

Anyways, I could just include my hardware info along with the benchmark. Benchmarks are still useful, especially when the variables are all controlled. 

Thank you again for your time and advice.
